data-exporter/src/main/java/org/reactome/release/dataexport/UniProtReactomeEntry.java

```diff
@@ -19,12 +20,18 @@
  * @author jweiser
  */
 public class UniProtReactomeEntry implements Comparable<UniProtReactomeEntry> {
+	private final Pattern PARENT_ACCESSION_REGEX = Pattern.compile(
+		"([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2})"
+	); // As defined at https://www.uniprot.org/help/accession_numbers (October 2019)
+
 	private static Map<Long, UniProtReactomeEntry> uniProtReactomeEntryMap = new HashMap<>();
 
-	private static Map<Session, Map<String, Set<ReactomeEvent>>> uniprotAccessionToTopLevelPathwaysCache =
-		new HashMap<>();
-	private static Map<Session, Map<String, Set<ReactomeEvent>>> uniprotAccessionToReactomeEventCache = new HashMap<>();
-	private static Map<Session, Map<String, Set<Long>>> uniprotAccessionToReactionLikeEventIdCache = new HashMap<>();
+	private static Map<Session, Map<UniProtReactomeEntry, Set<ReactomeEvent>>>
```

Comment:
I might just not have got far enough in the PR, but where does UniProtReactomeEntry and ReactomeEvent data types come from? Are they a custom thing? What do they look like?

Reply:
I'm curious about why you decided to implement custom objects? As always, my focus is on the future developer who will come to Reactome to look at the codebase. I do think this could be a clarifying feature, but I am just curious about the motivation

Reply:
I needed classes to hold and manipulate data for Reactome events (pathways and reaction like events) and Reference Gene Products for UniProt instances.  The Java API and data model has some of that functionality but it made sense to create these two classes to hold the specific data and methods needed for the project.  I may re-organize the classes in this project into more specific subpackages going forward so the project structure is a bit clearer.